student_explorer/cron.py

```diff
@@ -0,0 +1,168 @@
+# -*- coding: future_fstrings -*-
+
+from __future__ import print_function #python 3 support
+
+from django.db import connections as conns
+
+import logging, datetime, json
+
+from django.conf import settings
+
+from django_cron import CronJobBase, Schedule
+
+from umich_api.api_utils import ApiUtil
+
+from seumich.models import Mentor
+
+import dateutil.parser as parser
+
+from decouple import config
+
+
+logger = logging.getLogger(__name__)
+
+# use ApiUtil for all API directory calls
+apiUtil = ApiUtil(config("APIUTIL_URL", default=""), config("APIUTIL_KEY", default=""), config("APIUTIL_SECRET", default=""))
+
+
+# cron job to populate course and user tables
+class StudentExplorerCronJob(CronJobBase):
+
+    schedule = Schedule(run_at_times=settings.RUN_AT_TIMES)
+    code = 'student_explorer.StudentExplorerCronJob'    # a unique code
+
+    def get_mentor_canvas_id(self, mentor_uniqname):
+        """
+        This process gets the canvas user id for mentor
+        :param mentor_uniqname:
+        :return:
+        """
+        mentor_result = apiUtil.api_call(f"aa/CanvasReadOnly/users/sis_login_id:{mentor_uniqname}/profile", "canvasreadonly")
+        mentor_json = json.loads(mentor_result.text)
+        return mentor_json.get("id", "")
+
+    def mentor_in_athletic(self, mentor_uniqname):
+        """
+        checks whether the mentor is in athletic department
+        :param mentor_uniqname:
+        :return:
+        """
+        in_athletic = False
+
+        logger.info("mentor uniquename=" + mentor_uniqname)
+        resp = apiUtil.api_call(f"MCommunity/People/{mentor_uniqname}", "mcommunity")
+        person_json = json.loads(resp.text)
```

Comment:
This will be improved in a future version of the apiUtil but this is good for now.